### PR TITLE
Preparations for federated dev

### PR DESCRIPTION
**Pull request by yoav-lavi**
# Description

 - Removes analytics
 - Refactors platform data (URLs and credentials)
 - Allows logging into instances other than the hosted instance
 - Adds a `--url` flag for `login`

### CODE REVIEW

File: cli/crates/common/src/environment.rs [hunk 1]
```diff
@@ -60,7 +94,8 @@ impl Environment {
                 user_dot_grafbase_path,
                 warnings: Vec::new(),
             })
-            .expect("cannot set environment twice");
+            .map_err(|_| ())
+            .expect("cannot) set environment twice");
```
Comment (tomhoule) [inline cli/crates/common/src/environment.rs:98]: small typo here ("cannot)")

File: workspace-hack/Cargo.toml [hunk 2]
```diff
@@ -26,6 +26,7 @@ aho-corasick = { version = "1" }
 anyhow = { version = "1" }
```
Comment (tomhoule) [inline workspace-hack/Cargo.toml:26]: Does this file need to be updated again, after the federated-dev re-removal?

**Review by tomhoule — APPROVED** [review]
The PlatformData refactoring is a nice improvement :100:

File: cli/crates/common/src/environment.rs (hunk 1 above)
Comment (yoav-lavi) [reply cli/crates/common/src/environment.rs:98]: I swear I've fixed this three different times now, I'm not sure why it keeps showing up haha

File: workspace-hack/Cargo.toml (hunk 2 above)
Comment (yoav-lavi) [reply workspace-hack/Cargo.toml]: Yeah, done

**Pull request merged**